src/Mimi.jl

```diff
@@ -215,17 +232,24 @@ function addcomponent(m::Model, t, name::Symbol=Symbol(string(t)); before=nothin
 
     if before!=nothing
         newcomponents = OrderedDict{Symbol,ComponentState}()
+        newcomponents2 = OrderedDict{Symbol, ComponentInstanceInfo}()
         for i in keys(m.components)
             if i==before
                 newcomponents[name] = comp
+                this = ComponentInstanceInfo(name, t)
```

Comment:
Maybe use a different name here than ``this``. That term is reserved in lots of programming languages, and who knows, maybe the julia people want to use it at some point as well.